meteor/lib/api/studios.ts

```diff
@@ -32,6 +33,6 @@ export interface UIStudio {
 
 	settings: IStudioSettings
 
-	routeSets: Record<string, StudioRouteSet>
-	routeSetExclusivityGroups: Record<string, StudioRouteSetExclusivityGroup>
+	routeSets: ObjectWithOverrides<Record<string, StudioRouteSet>>
```

Comment:
These shouldn't be using `ObjectWithOverrides`.
The intention of these `UI....` publications is to do some light preprocessing of some collections before sending them into the ui. Primarily intended to strip out the `ObjectWithOverrides`, so that the ui doesn't need to run `applyAndValidateOverrides` and try to memoize/cache it sufficiently to avoid re-renders when nothing changes.

The `mappings` property on this interface is already doing this, you can follow the pattern it uses to handle this inside of the publication.

Reply:
Ok, that makes sense, I'll do it as mapping and then we can revert the Switchboard and the RundownRighthanControl.
Should we then also rename the current backend RouteSets to RouteSetsWithOverrides and the UI version to RouteSetExt as done with mapping?

Reply:
Fixed